libs/util/http.client.js

```diff
@@ -200,111 +135,81 @@ function FetchrError(options, request, response, responseBody, originalError) {
+        timeout: options.timeout,
+        url: options.url,
+    }).catch(function (err) {
+        if (!shouldRetry(options, err.statusCode, currentAttempt)) {
+            throw err;
+        }
+
+        // Use exponential backoff and full jitter
+        // strategy published in
+        // https://aws.amazon.com/blogs/architecture/exponential-backoff-and-jitter/
+        var delay =
+            Math.random() *
+            options.retry.interval *
+            Math.pow(2, currentAttempt);
+
+        return delayPromise(function () {
+            return httpRequest(rawOptions, currentAttempt + 1);
+        }, delay);
+    });
+
+    return {
+        then: promise.then.bind(promise),
+        catch: promise.catch.bind(promise),
+        abort: controller.abort.bind(controller),
+    };
 }
 
-/**
- * @class REST.HTTP
- */
-module.exports = {
-    /**
-     * @method get
-     * @public
-     * @param {String} url
-     * @param {Object} headers
-     * @param {Object} config  The config object.
-     * @param {Number} [config.timeout=3000] Timeout (in ms) for each request
-     * @param {Object} config.retry Retry config object.
-     * @param {Number} [config.retry.interval=200] The start interval unit (in ms).
-     * @param {Number} [config.retry.maxRetries=0] Number of max retries.
-     * @param {Number} [config.retry.statusCodes=[0, 408, 999]] Response status codes to be retried.
-     * @param {Boolean} [config.cors] Whether to enable CORS & use XDR on IE8/9.
-     * @param {Function} callback The callback function, with two params (error, response)
-     */
-    get: function (url, headers, config, callback) {
-        return doRequest(
-            METHOD_GET,
-            url,
-            headers,
-            null,
-            config,
-            INITIAL_ATTEMPT,
-            callback
-        );
-    },
-
-    /**
-     * @method post
-     * @param {String} url
-     * @param {Object} headers
-     * @param {Mixed}  data
-     * @param {Object} config  The config object. No retries for POST.
-     * @param {Number} [config.timeout=3000] Timeout (in ms) for each request
-     * @param {Boolean} [config.unsafeAllowRetry=false] Whether to allow retrying this post.
-     * @param {Number} [config.retry.interval=200] The start interval unit (in ms).
-     * @param {Number} [config.retry.maxRetries=0] Number of max retries.
-     * @param {Number} [config.retry.statusCodes=[0, 408, 999]] Response status codes to be retried.
-     * @param {Boolean} [config.cors] Whether to enable CORS & use XDR on IE8/9.
-     * @param {Function} callback The callback function, with two params (error, response)
-     */
-    post: function (url, headers, data, config, callback) {
-        return doRequest(
-            METHOD_POST,
-            url,
-            headers,
-            data,
-            config,
-            INITIAL_ATTEMPT,
-            callback
-        );
-    },
-};
+module.exports.default = httpRequest;
```

Comment:
No big reason behind it. I names as default since probably in the future, this line of code will become: `export default httpRequest`. 

How would you prefer? We can export it as `httpRequest` (or any other name). The only thing that I'm afraid is to export it directly (`module.exports = httpRequest`) since it might break the tests because of the mocks.

Reply:
ok we can leave for now, once if/when move to TS, these will all becomes import/export anyways